messages/src/lib.rs

```diff
@@ -1,35 +1,84 @@
+pub mod chunking;
+
+use anyhow::{bail, Result};
+use cid::multihash::{Code, MultihashDigest};
+use cid::Cid;
 use clap::Subcommand;
-use parity_scale_codec::Encode;
+use parity_scale_codec::{Decode, Encode};
 use parity_scale_codec_derive::{Decode as ParityDecode, Encode as ParityEncode};
 use serde::Serialize;
-// use std::collections::BTreeMap;
 
-#[derive(Debug, ParityEncode, ParityDecode, Serialize)]
+// This MessageContainer struct is intended to be used inside of the chunkers
+// for verification of Message integrity during the chunking/assembly process
+// Also all Messages are now transferred in IPLD blocks *tada*
+#[derive(Clone, Debug, ParityEncode, ParityDecode, Serialize, Eq, PartialEq)]
+pub(crate) struct MessageContainer {
+    // CID using hash of payload (pre-serialized as Vec<u8>)
+    cid: Vec<u8>,
+    // Message payload
+    pub message: Message,
+}
+
+impl MessageContainer {
+    pub fn new(message: Message) -> Self {
+        let hash = Code::Blake2s128.digest(&message.to_bytes());
```

Comment:
Are we using 128 bits (instead of the [normal hash size](https://proto.school/anatomy-of-a-cid/01)) due to size limitations?

Reply:
Yup that is correct. I wanted a hash for message verification on the other side, but I also wanted to limit the extra overhead added by the hash, so this seemed like a reasonable way to accomplish that.

Reply:
Would you mind adding this note somewhere?